pallets/loan/src/benchmarking.rs

```diff
@@ -0,0 +1,528 @@
+	<T as LoanConfig>::Rate: From<Rate>,
+	<T as LoanConfig>::Amount: From<Amount>,
+{
+	let loan_type = LoanType::BulletLoan(BulletLoan::new(
+		// advance rate 80%
+		Rate::saturating_from_rational(80, 100).into(),
+		// expected loss over asset maturity 0.15%
+		Rate::saturating_from_rational(15, 10000).into(),
+		// collateral value
+		Amount::from_inner(125 * CFG).into(),
+		// 4%
+		math::rate_per_sec(Rate::saturating_from_rational(4, 100))
+			.unwrap()
+			.into(),
+		// 2 years
+		math::seconds_per_year() * 2,
+	));
+	// interest rate is 5%
+	let rp: T::Rate = math::rate_per_sec(Rate::saturating_from_rational(5, 100))
+		.unwrap()
+		.into();
+	let origin = T::AdminOrigin::successful_origin();
+	LoanPallet::<T>::activate_loan(origin, pool_id, loan_id, rp, loan_type)
+		.expect("loan activation should not fail");
+}
+
+fn add_test_write_off_groups<T: Config>(pool_id: PoolIdOf<T>)
+where
+	<T as LoanConfig>::Rate: From<Rate>,
+{
+	for group in vec![(3, 10), (5, 15), (7, 20), (20, 30), (120, 100)] {
+		LoanPallet::<T>::add_write_off_group_to_pool(
+			T::AdminOrigin::successful_origin(),
+			pool_id,
+			WriteOffGroup {
+				percentage: Rate::saturating_from_rational(group.1, 100).into(),
+				overdue_days: group.0,
+			},
+		)
+		.expect("adding write off groups should not fail");
+	}
+}
+
+benchmarks! {
+	where_clause {
+		where
+		<T as pallet_uniques::Config>::ClassId: From<u64>,
+		<T as pallet_balances::Config>::Balance: From<u128>,
+		CurrencyIdOf<T>: From<u32>,
+		<T as LoanConfig>::Rate: From<Rate>,
+		<T as LoanConfig>::Amount: From<Amount>,
+		PoolIdOf<T>: From<<T as pallet_pool::Config>::PoolId>,
+		<T as ORMLConfig>::Balance: From<u128>,
+		<T as ORMLConfig>::CurrencyId: From<u32>,
+		<T as TimestampConfig>::Moment: From<u64> + Into<u64>,
+	}
+
+	initialise_pool {
+		let origin = T::AdminOrigin::successful_origin();
+		let pool_id: PoolIdOf<T> = Default::default();
+		let class_id: <T as LoanConfig>::ClassId = Default::default();
+		let call = Call::<T>::initialise_pool(pool_id, class_id);
+	}:{ call.dispatch_bypass_filter(origin)? }
+	verify {
+		let got_class_id = PoolToLoanNftClass::<T>::get(pool_id).expect("pool must be initialised");
+		assert_eq!(class_id, got_class_id);
+		let got_pool_id = LoanNftClassToPool::<T>::get(got_class_id).expect("nft class id must be initialised");
+		assert_eq!(pool_id, got_pool_id);
+	}
+
+	issue_loan {
+		let (pool_owner, pool_id, loan_account, loan_class_id) = create_and_init_pool::<T>();
+		let (loan_owner, asset) = create_asset::<T>();
+	}:_(RawOrigin::Signed(loan_owner.clone()), pool_id, asset)
+	verify {
+		// assert loan issue event
+		let loan_id: T::LoanId = 1u128.into();
+		assert_last_event::<T>(LoanEvent::LoanIssued(pool_id, loan_id, asset).into());
+
+		// asset owner must be loan account
+		expect_asset_owner::<T>(asset, loan_account);
+
+		// loan owner must be caller
+		let loan_asset = Asset(loan_class_id, loan_id);
+		expect_asset_owner::<T>(loan_asset, loan_owner);
+	}
```

Comment:
yeah. dont feel like bloating the benchmarks on storage level since we only care about the event level checks. This has extensive unit tests that cover storage  checks as well